models/model.php

```diff
@@ -245,13 +245,23 @@ function event_calendar_get_events_between($start_date, $end_date, $is_count=fal
 
 function event_calendar_merge_repeating_events($events, $repeating_events) {
 	$non_repeating_events = array();
-	foreach($events as $e) {
-		if ($e->repeats != 'yes') {
-			$non_repeating_events[] = array('event' => $e,'data' => array(array('start_time' => $e->start_date, 'end_time' => $e->real_end_time)));
+	if (is_array($events) && count($events) > 0) {
+		foreach($events as $e) {
+			if ($e->repeats != 'yes') {
+				$non_repeating_events[] = array('event' => $e,'data' => array(array('start_time' => $e->start_date, 'end_time' => $e->real_end_time)));
+			}
 		}
+		if (is_array($repeating_events) && count($repeating_events) > 0) {
+			if (count($non_repeating_events) > 0) {
+				return array_merge($non_repeating_events, $repeating_events);
+			} else {
+				return $repeating_events;
+			}
+		}
+	} else if (is_array($repeating_events) && count($repeating_events) > 0) {
+		return $repeating_events;
 	}
```

Comment:
Could you add some comments to describe what is happening in this function and why?